generated/nidcpower/nidcpower/session.py

```diff
@@ -4846,11 +4862,27 @@ def measure_multiple(self):
 
         '''
         import collections
-        Measurement = collections.namedtuple('Measurement', ['voltage', 'current', 'in_compliance'])
+        Measurement = collections.namedtuple('Measurement', ['voltage', 'current', 'in_compliance', 'channel'])
 
         voltage_measurements, current_measurements = self._measure_multiple()
 
-        return [Measurement(voltage=voltage_measurements[i], current=current_measurements[i], in_compliance=None) for i in range(self._parse_channel_count())]
+        if self._repeated_capability == '':
```

Comment:
> This if/else statement should probably be factored out into own function.

@marcoskirsch Where should this "if/else" statement live? In `_SessionBase`? In the `_converters`?

If it is to be added into `_SessionBase`, how should I make sure that it is generated for nidcpower only? Should I add an if/else to the `session.py.mako` template? Or add it as a function in the metadata and use another template (e.g. `fancy_fetch.py.mako`) to store its implementation?

>  What is it doing exactly? Setting a list to channel_names?

Yes, it is creating a list of channel names based on the repeated capabilities so that the list can be used to populate the `channel` member of the `Measurement` namedtuple.

Reply:
We could push the if/else logic down to `convert_channels_repeated_capabilities`, especially given the function would need access to "all channels in the session" to handle the 2 TBDs mentioned [here](https://github.com/ni/nimi-python/pull/1754#discussion_r912223170).

Regarding getting and storing "all channels in the session", I was thinking we add a function to `_SessionBase` called `_get_session_channels`, in where we call `_get_channel_names`. We would need to surround the driver runtime function call with `try/catch` since not all driver runtimes support it.
Additionally we could use the `@lru_cache` decorator to ensure we do the actual work only once.

Reply:
> We could push the if/else logic down to `convert_channels_repeated_capabilities`, especially given the function would need access to "all channels in the session" to handle the 2 TBDs mentioned https://github.com/ni/nimi-python/pull/1754#discussion_r912223170.

@sbethur Do you mean I should change the function signature of `convert_channels_repeated_capabilities()` to something like
```
def convert_channels_repeated_capabilities(
    channels_repeated_capability,
    session_channel_names
):
```
and call if from `measure_multiple()` like
```
channel_names = _converters.convert_channels_repeated_capabilities(
    self._repeated_capability,
    self._get_channel_names(range(self.channel_count))
)
```
?

Please note that I've replied to your comment regarding the 2 TBDs at https://github.com/ni/nimi-python/pull/1754#discussion_r912875369.

> Regarding getting and storing "all channels in the session"

Can you clarify what do you mean by "all channels in the session"? Do you mean channels that are opened as part of the Session constructor? Isn't that what `get_channel_names()` is currently doing?

If you mean channels that are specified by `session.channels[...]`, isn't that what my `convert_channels_repeated_capabilities()` is trying to achieve here?

Reply:
I have made the change for `convert_channels_repeated_capabilities()` to allow pushing the if/else logic from the "fancy fetch" body to the converter function in my latest commit:
https://github.com/ni/nimi-python/pull/1754/commits/8641fc8cd4031be0a04f86f97f575dc470fea3a3

Reply:
> Can you clarify what do you mean by "all channels in the session"? Do you mean channels that are opened as part of the Session constructor? Isn't that what get_channel_names() is currently doing?

Yes, "channels that are opened as part of the Session constructor". 

>  Isn't that what get_channel_names() is currently doing?
It is. This data is constant throughout the session. So we could fetch it from the driver runtime once and store it as class state.